spring-integration-core/src/main/java/org/springframework/integration/channel/AbstractMessageChannel.java

```diff
@@ -173,16 +175,28 @@ public void setDatatypes(Class<?>... datatypes) {
 	public void setInterceptors(List<ChannelInterceptor> interceptors) {
 		Collections.sort(interceptors, this.orderComparator);
 		this.interceptors.set(interceptors);
+		populateBeanFactoryToInterceptors(interceptors.toArray(new ChannelInterceptor[interceptors.size()]));
+	}
+
+	protected void populateBeanFactoryToInterceptors(ChannelInterceptor... interceptors) {
```

Comment:
Yes, I received an email from github.

I think the wiretap should properly be a bean, just like it is from the XML parser.

I am not keen on having the abstract channel making changes to the interceptor.